calblueprint/meepanyar
# InventoryData setup + Add Inventory Functionality

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
- Added setup for the `inventoryData` slice.
- Basic inventory setup -- the main list of inventory works with cards that link to (mostly empty) inventory profiles
- Working functionality to add an inventory item (aka add a product to a site). The list of options only shows the products that aren't already in the site

## How to review
To test the add functionality, run `annie/inventory-items` on the backend ([companion PR](https://github.com/calblueprint/meepanyar-node/pull/16))
- Navigate to the inventory tab (shopping cart)
- Click the **+** button
- Try adding inventory and see if it:
  - limits selection to products that the site doesn't already have
  - adds the new inventory record to redux
  - shows up in Airtable correctly
  - navigates to the new profile page corresponding to the correct inventory item + qty
  
Note: if there aren't any choices in the dropdown, it may be because the site already carries every product, so you'll have to delete them from Airtable. I think we need to eventually address this edge case through the designs.
  
**Switching between sites**
- In `testUser`'s account, select the East Asian Library site and add an inventory item
- Switch to Kyle's dancing places and check that the inventory list has changed
- Add an inventory item to Kyle's dancing places and check that it shows up properly.

## Relevant Links

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

- https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype
- https://material-ui.com/components/autocomplete/

### Related PRs
Corresponding backend PR: https://github.com/calblueprint/meepanyar-node/pull/16

## Next steps
- Automatically trigger an inventory update when creating a new inventory item
- Add input validation and error messaging
- Add the ability to add new products
- Add relevant details to inventory profile + inventory card to match designs
- Clean up/remove data we don't use (e.g. `inventoryIds` in `currentSite` and products) 

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
![image](https://user-images.githubusercontent.com/21160510/111921481-0f164080-8a52-11eb-80e3-db128afcfbb7.png)
![image](https://user-images.githubusercontent.com/21160510/111921492-1d645c80-8a52-11eb-8b38-2fd31a78129d.png)

CC: @julianrkung

[//]: # 'This tags Julian as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
